### PR TITLE
Bump version to v0.3.0

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Bump Ruff to v0.3

### CODE REVIEW

**Comment by Skylion007** [comment]
We aren't promoting any rules from preview in this release?

**Comment by MichaReiser** [comment]
This is not ready to merge but ready to get early feedback on the new documentation

**Comment by Skylion007** [comment]
@charliermarsh Thoughts on the PERF lint rules?

File: crates/ruff_python_formatter/README.md [hunk 1]
```diff
@@ -606,44 +742,3 @@ df.drop(columns=["aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"]).drop_duplicates(a).rename
     }
 ).to_csv(path / "aaaaaa.csv", index=False).other(a)
 ```
-
-#### Expressions with (non-pragma) trailing comments are split more often ([#7823](https://github.com/astral-sh/ruff/issues/7823))
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/README.md:610]: We fixed this at some point

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,68 @@
...
+
+- Explicitly disallow `extend` as part of a `--config` flag ([#10135](https://github.com/astral-sh/ruff/pull/10135))
+- Remove `build` from the default exclusion list ([#10093](https://github.com/astral-sh/ruff/pull/10093))
+- Deprecate `ruff <path>`, `ruff --explain`, `ruff --clean`, and `ruff --generate-shell-completion` in favor of `ruff check <path>`, `ruff rule`, `ruff clean`, and `ruff generate-shell-completion` ([#10169](https://github.com/astral-sh/ruff/pull/10169))
+
+### Bug fixes
+
+- \[`pycodestyle`\] Mark fixes overlapping with a multiline string as unsafe (`W293`) ([#10049](https://github.com/astral-sh/ruff/pull/10049))
+- \[`flake8-type-checking`\] Respect runtime-required decorators for function signatures ([#10091](https://github.com/astral-sh/ruff/pull/10091))
+- Set PowerPC page size to 64KB ([#10080](https://github.com/astral-sh/ruff/pull/10080))
```
[the remark this answers is not in the document]
Comment (MichaReiser) [reply CHANGELOG.md:51]: I can rephrase it to explain the issue it fixes

[the remark this answers is not in the document]
Comment (AlexWaygood) [reply CHANGELOG.md:51]: that would be great!

File: crates/ruff_python_formatter/README.md [hunk 3]
```diff
@@ -208,37 +208,167 @@ on both `first()` and `second()`:
...
+if x:
+
+  a = 123
+
+# Ruff
+if x:
+  a = 123
+```
+
+Currently, we are concerned that allowing blank lines at the start of a block leads [to unintentional blank lines when refactoring or moving code](https://github.com/astral-sh/ruff/issues/8893#issuecomment-1867259744). However, we will consider adotping Black's formatting at a later point with an improved heuristic. The style change is tracked in [#9745](https://github.com/astral-sh/ruff/issues/9745).
```
Comment (AlexWaygood) [inline crates/ruff_python_formatter/README.md:328]: I think I'm to blame for this one...

```suggestion
Currently, we are concerned that allowing blank lines at the start of a block leads [to unintentional blank lines when refactoring or moving code](https://github.com/astral-sh/ruff/issues/8893#issuecomment-1867259744). However, we will consider adopting Black's formatting at a later point with an improved heuristic. The style change is tracked in [#9745](https://github.com/astral-sh/ruff/issues/9745).
```

**Review by AlexWaygood — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,68 @@
...
+- \[`ruff`\] Avoid f-string false positives in `gettext` calls (`RUF027`) ([#10118](https://github.com/astral-sh/ruff/pull/10118))
+- \[`pylint`\] Delete entire statement, including semicolons (`PLR0203`) ([#10074](https://github.com/astral-sh/ruff/pull/10074))
```
[the remark this answers is not in the document]
Comment (MichaReiser) [reply CHANGELOG.md:53]: I keep the `(rule)` for consistency with other changelog entries
